pkg/resources/grant_application_role_acceptance_test.go

```diff
@@ -0,0 +1,142 @@
+					resource.TestCheckResourceAttr(resourceName, "id", fmt.Sprintf(`"%v"."%v"|ACCOUNT_ROLE|"%v"`, applicationName, applicationRoleName, parentAccountRoleName)),
+				),
+			},
+			// test import
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				ConfigDirectory:          config.StaticDirectory("testdata/TestAcc_GrantApplicationRole/account_role"),
+				ConfigVariables:          m(),
+				ResourceName:             resourceName,
+				ImportState:              true,
+				ImportStateVerify:        true,
+			},
+		},
+	})
+}
+
+func TestAcc_GrantApplicationRole_application(t *testing.T) {
+	applicationName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	applicationName2 := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	resourceName := "snowflake_grant_application_role.g"
+	applicationRoleName := "app_role_1"
+	applicationRoleNameFullyQualified := fmt.Sprintf("\"%s\".\"%s\"", applicationName, applicationRoleName)
+	randomName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+
+	m := func() map[string]config.Variable {
+		return map[string]config.Variable{
+			"database_name":     config.StringVariable(acc.TestDatabaseName),
+			"schema_name":       config.StringVariable(acc.TestSchemaName),
+			"application_name":  config.StringVariable(applicationName),
+			"application_name2": config.StringVariable(applicationName2),
+			"random_name":       config.StringVariable(randomName),
+		}
+	}
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: testAccCheckGrantApplicationRoleDestroy,
+		Steps: []resource.TestStep{
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				ConfigDirectory:          config.StaticDirectory("testdata/TestAcc_GrantApplicationRole/application"),
+				ConfigVariables:          m(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "application_role_name", applicationRoleNameFullyQualified),
+					resource.TestCheckResourceAttr(resourceName, "application_name", fmt.Sprintf("\"%s\"", applicationName2)),
+					resource.TestCheckResourceAttr(resourceName, "id", fmt.Sprintf(`"%v"."%v"|APPLICATION|"%v"`, applicationName, applicationRoleName, applicationName2)),
+				),
+			},
+			// test import
+			{
+				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+				ConfigDirectory:          config.StaticDirectory("testdata/TestAcc_GrantApplicationRole/application"),
+				ConfigVariables:          m(),
+				ResourceName:             resourceName,
+				ImportState:              true,
+				ImportStateVerify:        true,
+			},
+		},
+	})
+}
+
+func testAccCheckGrantApplicationRoleDestroy(s *terraform.State) error {
```

Comment:
we have a separate place for the checkDestroy functions now (pkg/acceptance/check_destroy.go)

Reply:
Moved.